### PR TITLE
Onboarding Riverpod Refactor

### PR DESCRIPTION
**Pull request by KalervoHyyppa**
## Description

This is a Refactor of Onboarding, adding the use of Riverpod

[CORRESPONDING PR IN RIBN-TOOLKIT](https://github.com/Topl/ribn-toolkit/pull/56)

Fixes # (issue)

[RIBN-673](https://topl.atlassian.net/jira/software/c/projects/RIBN/boards/5?modal=detail&selectedIssue=RIBN-673)

## Checklist:

-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] I have made corresponding changes to the documentation
-   [x] My changes generate no new warnings
-   [x] I have added tests that prove my fix is effective or that my feature works
-   [x] New and existing unit tests pass locally with my changes
-   [x] Any dependent changes have been merged and published in downstream modules
-   [x] I have checked my code and corrected any misspellings

[RIBN-673]: https://topl.atlassian.net/browse/RIBN-673?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: lib/utils/error_handling_utils.dart [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import 'package:ribn/constants/routes.dart';
+import 'package:ribn/utils/navigation_utils.dart';
+
+/// This will handle an error and navigate to an error page
+Future<void> handleApiError({
```
Comment (ultimaterex) [inline lib/utils/error_handling_utils.dart:5]: I love this, would there be any added value to log a warning or something similar to the console here?

File: test/mocks/local_authentication_mocks.mocks.dart [hunk 2]
```diff
@@ -0,0 +1,87 @@
+// Mocks generated by Mockito 5.3.2 from annotations
```
Comment (ultimaterex) [inline test/mocks/local_authentication_mocks.mocks.dart:1]: Ditto about generated files being committed to the repo

**Review by ultimaterex — CHANGES_REQUESTED** [review]
Content wise, it seems solid and I have not seen any major issues. I do have a bunch of styling nitpicks, I have yet to run a physical test. I'll update you with that feedback once you've addressed my comments.

File: lib/models/keychain_state.dart [hunk 3]
```diff
@@ -34,6 +34,7 @@ class KeychainState {
   /// Gets all supported [RibnNetwork]s.
```
[the remark this answers is not in the document]
Comment (ultimaterex) [reply lib/models/keychain_state.dart:34]: specifically the //Todo

**Comment by ultimaterex** [comment]
The linter in general is also quite unhappy about some nullable marked members that don't return values

File: lib/presentation/onboarding/create_wallet/seed_phrase_confirmation_page.dart [hunk 4]
```diff
@@ -13,114 +16,118 @@ import 'package:ribn/constants/assets.dart';
...
 import 'package:ribn/presentation/onboarding/widgets/confirmation_button.dart';
 import 'package:ribn/presentation/onboarding/widgets/onboarding_container.dart';
 import 'package:ribn/presentation/onboarding/widgets/web_onboarding_app_bar.dart';
 import 'package:ribn/utils.dart';
 
-class SeedPhraseConfirmationPage extends StatefulWidget {
-  const SeedPhraseConfirmationPage({Key? key}) : super(key: key);
+class SeedPhraseConfirmationPage extends HookConsumerWidget {
+  static const Key seedPhraseConfirmationPageKey = Key('seedPhraseConfirmationPageKey');
+  static const Key seedPhraseConfirmationConfirmationButtonKey =
```
[the remark this answers is not in the document]
Comment (KalervoHyyppa) [reply lib/presentation/onboarding/create_wallet/seed_phrase_confirmation_page.dart:27]: I generally like to keep them in their widgets (or parent widgets) so it's easier to find where exactly they are being used vs having to search mutliple files.

File: lib/models/keychain_state.dart (hunk 3 above)
[the remark this answers is not in the document]
Comment (KalervoHyyppa) [reply lib/models/keychain_state.dart:34]: I've made tickets for all models to utilize freezed. [RIBN-697](https://topl.atlassian.net/browse/RIBN-697), specifically for this one. I'll remove the todo

**Comment by KalervoHyyppa** [comment]
> The linter in general is also quite unhappy about some nullable marked members that don't return values

@ultimaterex Regarding analyzer, I'm seeing 2 reasons why errors are being thrown,

1. Autogenerated files have been removed, so there are errors because of that
2. ribn-toolkit needs to be updated

**Comment by ultimaterex** [comment]
> > The linter in general is also quite unhappy about some nullable marked members that don't return values
> 
> @ultimaterex Regarding analyzer, I'm seeing 2 reasons why errors are being thrown,
> 
> 1. Autogenerated files have been removed, so there are errors because of that
> 2. ribn-toolkit needs to be updated

Re 1: We should make an action item to add the make gen step to the CI analyzer so that it can successfully run the tests.

**Review by ultimaterex — APPROVED** [review]

**Comment by ultimaterex** [comment]
I've approved the style changes for this PR and will let you know how manual testing goes.

File: lib/providers/onboarding_provider.dart [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+
+  static String _generateMnemonic(Ref ref) {
+    final Random random = ref.read(randomProvider)();
+
+    final entropy = ref.read(entropyProvider)(random);
+    return ref.read(entropyFuncProvider)(entropy);
+  }
+
+  regenerateMnemonic() {
+    final onboardingState = _generateOnboardingState(ref);
```
Comment (Brianspha) [inline lib/providers/onboarding_provider.dart:47]: Here we could be explicit with the type of onboardingState
e.g.
final OnboardingState onboardingState = _generateOnboardingState(ref);

**Review by Brianspha — CHANGES_REQUESTED** [review]
Additionally, please see lines 150 on the enable_biometrics.dart page please remove the 

     ```
       const SizedBox(
              height: 15,
            ),
```

On the same file please change the font size of the heading (Enable Biometrics) to 20

**Review by Brianspha — APPROVED** [review]

**Pull request merged**